src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -419,6 +450,67 @@ private string GetSymbolToEmitForTargetPlatform(string symbol)
             return symbolToEmit;
         }
 
+        public int EmitSymbolReference(ISymbolNode target, int delta, RelocType relocType)
+        {
+            string targetName = GetSymbolToEmitForTargetPlatform(target.MangledName);
+            int size = 0;
+            bool isPCRelative = false;
+            SymbolRefExpr kind = SymbolRefExpr.None;
+            switch (relocType)
+            {
+                case RelocType.IMAGE_REL_BASED_DIR64:
+                    size = 8;
+                    break;
+                case RelocType.IMAGE_REL_BASED_REL32:
+                    size = 4;
+                    isPCRelative = true;
+                    break;
+                case RelocType.IMAGE_REL_BASED_ABSOLUTE:
+                    size = 4;
+                    kind = SymbolRefExpr.COFF_IMGREL32;
+                    break;
+                default:
+                    throw new NotImplementedException();
+            }
+            // Emit symbol reference
+            EmitSymbolRef(targetName, size, isPCRelative, delta, kind);
+            return size;
+        }
+
+        public void EmitBlobWithRelocs(byte[] blob, Relocation[] relocs)
+        {
+            int nextRelocOffset = -1;
+            int nextRelocIndex = -1;
+            if (relocs.Length > 0)
+            {
+                nextRelocOffset = relocs[0].Offset;
+                nextRelocIndex = 0;
+            }
+
+            int i = 0;
+            while (i < blob.Length)
+            {
```

Comment:
There is a similar code duplicated in other place in this file. This will need to be refactored to reduce the per-byte overhead (https://github.com/dotnet/corert/issues/762), so I have not tried to make it pretty.